sys/riscv/riscv/pmap.c

```diff
@@ -4126,8 +4126,27 @@ pmap_caploadgen_update(pmap_t pmap, vm_offset_t va, vm_page_t *mp, int flags)
 
 	return res;
 }
-#endif
-#endif
+
+void
+pmap_assert_consistent_clg(pmap_t pmap, vm_offset_t va)
+{
+	pt_entry_t *pte, tpte;
+
+	pte = pmap_l3(pmap, va);
```

Comment:
Does this need to care about superpages and try the l2 first before an l2_to_l3? I know at one point there was talk of having to break down superpages during revocation so maybe not?

Reply:
I'm not sure. We do break them down during the sweep and and try to put them back together when we get to the end. I'm definitely less confident this code is correct than with the Arm version.

Reply:
Yes, I think @jrtc27 is correct that this needs to look for a leaf L2 PTE before insisting on a L3 PTE.

Reply:
Should be fixed if I've read the rest of the code correctly. Not convinced we're exercising the L2 path so we might need to figure out how to add some tests.